### PR TITLE
Add release notes for 1.62

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
cc @rust-lang/release 

r? @pietroalbini

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,132 @@
...
+Compiler
+--------
+
+- [linker: Stop using whole-archive on dependencies of dylibs][96436]
+- [Make `unaligned_references` lint deny-by-default][95372]
+  This lint is also a future compatibility lint, and is expected to eventually
+  become a hard error.
+- [Only add codegen backend to dep info if -Zbinary-dep-depinfo is used][93969]
+- [Reject `#[thread_local]` attribute on non-static items][95006]
+- [Add tier 3 `aarch64-pc-windows-gnullvm` and `x86_64-pc-windows-gnullvm` targets\*]
```
Comment (wesleywiser) [inline RELEASES.md:23]:
```suggestion
- [Add tier 3 `aarch64-pc-windows-gnullvm` and `x86_64-pc-windows-gnullvm` targets\*][94872]
```

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,132 @@
...
+These changes provide no direct user facing benefits, but represent significant
+improvements to the internals and overall performance of rustc
+and related tools.
+
+[93313]: https://github.com/rust-lang/rust/pull/93313/
+[93969]: https://github.com/rust-lang/rust/pull/93969/
+[94079]: https://github.com/rust-lang/rust/pull/94079/
+[94206]: https://github.com/rust-lang/rust/pull/94206/
+[94457]: https://github.com/rust-lang/rust/pull/94457/
+[94775]: https://github.com/rust-lang/rust/pull/94775/
```
Comment (wesleywiser) [inline RELEASES.md:100]:
```suggestion
[94775]: https://github.com/rust-lang/rust/pull/94775/
[94872]: https://github.com/rust-lang/rust/pull/94872/
```

File: RELEASES.md [hunk 3]
```diff
@@ -1,3 +1,133 @@
...
+[96436]: https://github.com/rust-lang/rust/pull/96436/
+[96557]: https://github.com/rust-lang/rust/pull/96557/
+
+[`bool::then_some`]: https://doc.rust-lang.org/stable/std/primitive.bool.html#method.then_some
+[`f32::total_cmp`]: https://doc.rust-lang.org/stable/std/primitive.f32.html#method.total_cmp
+[`f64::total_cmp`]: https://doc.rust-lang.org/stable/std/primitive.f64.html#method.total_cmp
+[`Stdin::lines`]: https://doc.rust-lang.org/stable/std/io/struct.Stdin.html#method.lines
+[`impl<T: Default> Default for AssertUnwindSafe<T>`]: https://doc.rust-lang.org/stable/std/panic/struct.AssertUnwindSafe.html#impl-Default
+[`From<Rc<str>> for Rc<[u8]>`]: https://doc.rust-lang.org/stable/std/rc/struct.Rc.html#impl-From%3CRc%3Cstr%3E%3E
+[`From<Arc<str>> for Arc<[u8]>`]: https://doc.rust-lang.org/stable/std/rc/struct.Arc.html#impl-From%3CArc%3Cstr%3E%3E
```
Comment (ehuss) [inline RELEASES.md:126]: Unescaped square brackets cannot be used in link labels. These end up breaking the parser somewhat.  I suggest considering using something less literal, like `[rc-from-str-u8]`.

Also, the `Arc` link is broken, I believe it is in the `sync` module, not `rc`.

File: RELEASES.md [hunk 4]
```diff
@@ -1,3 +1,133 @@
...
+==========================
+
+Language
+--------
+
+- [Stabilize `#[derive(Default)]` on enums with a `#[default]` variant][94457]
+- [Stop validating some checks in dead code after functions with uninhabited return types][93313]
+- [Fix constants not getting dropped if part of a diverging expression][94775]
+- [Support unit struct/enum variant in destructuring assignment][95380]
+- [Allow inline consts to reference generic params][96557]
```
Comment (ehuss) [inline RELEASES.md:11]: Inline consts are nightly-only. I presume nightly-only things usually should not be included in the release notes?

Comment (Mark-Simulacrum) [reply RELEASES.md]: You're correct, removed the line.

File: RELEASES.md [hunk 5]
```diff
@@ -1,3 +1,131 @@
...
+
+\* Refer to Rust's [platform support page][platform-support-doc] for more
+   information on Rust's tiered platform support.
+
+Libraries
+---------
+
+- [Move `CStr` to libcore, and `CString` to liballoc][94079]
+- [Windows: Use a pipe relay for chaining pipes][95841]
+- [Use a single ReentrantMutex implementation on all platforms.][96042]
```
Comment (wwylele) [inline RELEASES.md:34]: Is `ReentrantMutex` exposed to end user? I don't find it in std documentation. If this is an internal type, maybe add more explanation for what this means for public types.

Comment (Mark-Simulacrum) [reply RELEASES.md:34]: Yeah, this is likely more of an internal change. I'll move it to that section -- thanks!

**Comment by Mark-Simulacrum** [comment]
Going to go ahead and merge this in since we're around a week and a half out from release; we can always apply any other changes in a separate PR.

@bors r+

**Comment by ehuss** [comment]
@Mark-Simulacrum FYI, this is not in the merge queue. Perhaps because it looks like there was a force push done immediately around the time of the `r+`?

**Comment by Mark-Simulacrum** [comment]
Oh, I probably squashed just after. @bors r+

**Pull request merged**